app/components/navigation/google-login-btn.tsx

```diff
@@ -0,0 +1,55 @@
+import Script from "next/script";
+import { useEffect, useState, useRef } from "react";
+
+interface GoogleLogin {
+    credential: string,
+    select_by: string
+}
+
+export default function GoogleLoginButton(){
+    const [scriptLoaded, setScriptLoaded] = useState(false);
+    const handleScriptLoad = () => {
+        setScriptLoaded(true);
+    };
+
+    const googleLoginRef = useRef(null);
+
+    useEffect(()=> {
+        if(scriptLoaded){
+            const google = window.google;
+            google.accounts.id.initialize({
+                client_id: "708154320268-432vdrg2g0h1652frag5vbu8r8qi4ers.apps.googleusercontent.com",
+                callback: handleCredentialResponse,
+                use_fedcm_for_prompt: true
+            });
+            const parent = googleLoginRef.current;
+            if(googleLoginRef && parent){
+                google.accounts.id.renderButton(parent, {
+                    text: "signin",
+                    size: "medium",
+                    logo_alignment: "left",
+                    shape: "rectangular",
+                    theme: "outline",
+                    type: "standard"
+                });
+            }
+        }
+    }, [scriptLoaded]);
+    
+    function handleCredentialResponse(response:GoogleLogin) {
+        fetch("/api/login", {
+            method: "POST",
+            body: JSON.stringify({ token: response.credential }),
+        }).then(handleLogin);
+    }
+
+    async function handleLogin(response: Response) {
+        const data = await response.json();
+        console.log(data);
+    }
+
+    return (<>
+        <div ref={googleLoginRef} id="google_login_btn"/>
+        <Script async src="https://accounts.google.com/gsi/client" onLoad={handleScriptLoad}/>
```

Comment:
Also, wondering should this be capitalized or should it be lower case? [These docs](https://react.dev/reference/react-dom/components/script) seem to have it lower case 🤔

Reply:
I don't know if it matters. I capitalized it to indicate it's the next script component as opposed to the HTML script tag (developer legibility).

Reply:
As we discussed off-line I do think it matters, but I guess the difference is if we are using the NextJS component or if we are using the standard HTML5 element. Which also as we discussed we are trying to use the NextJS component so uppercase is probably correct then!